packages/fiori-freestyle-writer/src/index.ts

```diff
@@ -34,15 +34,30 @@ async function generate<T>(basePath: string, data: FreestyleApp<T>, fs?: Editor)
     // Common files
     fs.copyTpl(join(tmplPath, 'common', 'add', '**/*.*'), basePath, ffApp);
 
-    fs.copyTpl(join(tmplPath, ffApp.template.type, 'add', `**/*.*`), basePath, ffApp, undefined, {
-        processDestinationPath(path: string): string {
-            if (ffApp.template.type === TemplateType.Basic) {
-                return path.replace('View1', (ffApp.template.settings as unknown as BasicAppSettings).viewName!);
-            } else {
-                return path;
-            }
-        }
-    });
+    fs.copyTpl(join(tmplPath, ffApp.template.type, 'add', `**/*.*`), basePath, ffApp, undefined, {});
+
+    if (ffApp.template.type === TemplateType.Basic) {
+        // Todo: need to find a way around the casts
```

Comment:
I don't think you need `as unknown` here.
`(ffApp.template.settings as BasicAppSettings).viewName` should be sufficient.